### PR TITLE
capture strerr separately for `build_container` command

### PR DESCRIPTION
**Pull request by hjmodi**
* CLOUDBLD-10938

Signed-off-by: Harsh Modi <hmodi@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/tasks/binary_container_build.py [hunk 1]
```diff
@@ -263,20 +266,21 @@ def build_container(
...
-            if line := build_process.stdout.readline():
-                yield line
-                # keep the last non-empty, non-whitespace line for an eventual error message
-                last_line = line.rstrip() or last_line
+        for line in build_process.stdout:
+            yield False, line
+        for line in build_process.stderr:
+            if line.rstrip():
+                error_lines.append(line)
+            yield True, line
```
Comment (MartinBasti) [inline atomic_reactor/tasks/binary_container_build.py:278]: does this work as expected, it doesn't mean that everyhing in STDERR is error

File: atomic_reactor/tasks/binary_container_build.py [hunk 2]
```diff
@@ -263,20 +266,21 @@ def build_container(
...
 
-            if (rc := build_process.poll()) is not None:
-                break
+        rc = build_process.poll()
 
         if rc != 0:
-            error = last_line if last_line else "<no output!>"
+            error = '\n'.join(error_lines) if error_lines else "<no output!>"
             raise BuildProcessError(f"Build failed (rc={rc}): {error}")
```
Comment (MartinBasti) [inline atomic_reactor/tasks/binary_container_build.py:284]: IMO, the aim of story was to just log this exception also as ERROR log line, to have it in log aggregation as ERROR

File: atomic_reactor/tasks/binary_container_build.py (hunk 1 above)
Comment (hjmodi) [reply atomic_reactor/tasks/binary_container_build.py]: changed to just take last line as error

File: atomic_reactor/tasks/binary_container_build.py (hunk 2 above)
Comment (hjmodi) [reply atomic_reactor/tasks/binary_container_build.py:303]: updated to use last line as error only.

**Review by mkosiarc — APPROVED** [review]

**Pull request merged**